android/titanium/src/java/org/appcelerator/titanium/io/TitaniumBlob.java

```diff
@@ -86,6 +91,19 @@ protected void init()
 					name = c.getString(0);
 					path = c.getString(1);
 				}
+
+				// must be a remote file, save locally as a temp file
+				// TODO: we should refactor our content resolving implementation
+				if (path == null) {
+					try {
+						InputStream inputStream =
+							TiApplication.getInstance().getContentResolver().openInputStream(Uri.parse(url));
+						File tempFile = TiFileHelper.getInstance().getTempFileFromInputStream(inputStream, name, true);
+						path = tempFile.getPath();
+					} catch (FileNotFoundException e) {
+						Log.e(TAG, "could not find " + url);
+					}
+				}
 			} finally {
```

Comment:
Not related to the ticket:
Would we want to catch exceptions when trying to get a cursor from the `query()` calls when the result is `null`? We can log a warning or an error message.